src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -690,29 +699,48 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     const parsedTracks = [];
     if (textTracks) {
       for (let i = 0; i < textTracks.length; i++) {
-        if (textTracks[i].language !== EXTERNAL_TRACK_ID || textTracks[i].label !== EXTERNAL_TRACK_ID) {
+        if (!Utils.textTrack.isExternalTrack(textTracks[i])) {
           const settings = {
             kind: textTracks[i].kind,
-            active: textTracks[i].mode === 'showing',
+            active: textTracks[i].mode === Utils.textTrack.SHOWING,
             label: textTracks[i].label,
             language: textTracks[i].language,
             index: i
           };
-          if (settings.kind === 'subtitles') {
+          if (settings.kind === Utils.textTrack.SUBTITLES) {
             parsedTracks.push(new PKTextTrack(settings));
             this._nativeTextTracksMap[settings.index] = textTracks[i];
-          } else if (settings.kind === 'captions' && this._config.enableCEA708Captions) {
+          } else if (settings.kind === Utils.textTrack.CAPTIONS && this._config.enableCEA708Captions) {
             settings.label = settings.label || captionsTextTrackLabels.shift();
             settings.language = settings.language || captionsTextTrackLanguageCodes.shift();
+            settings.available = this._captionsHidden;
             parsedTracks.push(new PKTextTrack(settings));
             this._nativeTextTracksMap[settings.index] = textTracks[i];
           }
         }
       }
+
+      if (!this._captionsHidden) {
```

Comment:
according to this `_captionsHidden` will always be hidden at first, but we are not checking if maybe the track has cues already which will mean it is already available

Reply:
Correct, there's no use case we're starting with cues or activeCues.
textTrack is `disabled` cues and activeCues removed from text track.
so it'll always start in `disabled` or `showing` because safari auto-selection but won't load any cue yet(right after load).
But there's a use case that new textTrack add, then we have an indicator for if caption still hidden